### DISCUSSION TITLE
GH Classroom Reuse Assignment -> 150+ unusable repos cannot be deleted!

### DISCUSSION THREAD

**Discussion by ADWolfe-LoyNO** [community/community#35264] [GitHub Education]
I too was happy about "reuse assignment," but after copying a dozen assignment, now my classroom repository has at least 150 repositories that are marked as "copying" and have been "copying" for at least three weeks.

I can't delete these zombies, gh CLI tells me:

`HTTP 403: Repository cannot be deleted until it is done being created on disk.`

This is a disaster in the making.

**Comment by samus-aran** [comment]
Hey @ADWolfe-LoyNO 👋 Let's see if we can assist. 

When you were using the reuse assignment was it done consecutively?
How quick would you say it happened? 
Is there any complexities to the assignments? Using autograding e.g
Are you able to share a screenshot of the interface showing the "copying" status for those repos.

Any additional information would be much appreciated and help assist the Classroom team on identifying the cause.

**Reply by ADWolfe-LoyNO** [reply]
`When you were using the reuse assignment was it done consecutively?`
I'm not sure what you mean by "consecutively." The "reuse assignment" function now presents the user with a list of assignments to reuse from the original classroom, which checkboxes to select one or more to reuse. When I did the "reuse assignment," I selected five or ten of these, gave my new organization and my new classroom, and clicked the button.

`How quick would you say it happened?`
I can't tell, my September was a bit of a nightmare. I hit the "reuse assignment" at one point and then  didn't get back to it for weeks. I can say that I tried another "reuse" and after ten minutes it had the "copying..." message still going.  These are small repos, no way they should take that long.

`Is there any complexities to the assignments? Using autograding e.g`
There is no autograding, but they all had "create feedback pull request." Most of them used the same template repository, which is derived from a repository I share with three other professors (we're at Loyola University New Orleans) for use with three of our programming courses, usually about 130 students each semester. This template has been in development for two or three years. 

`Are you able to share a screenshot of the interface showing the "copying" status for those repos.`

![image](https://user-images.githubusercontent.com/44787867/194932600-48e872de-376b-4146-ad91-1e57b4b6fa47.png)

also see:
![image](https://user-images.githubusercontent.com/44787867/194932750-8f22ee0e-e7e7-4a2f-9081-224014aae1a0.png)

`Any additional information would be much appreciated and help assist the Classroom team on identifying the cause.`

We have observed that templates and reused assignments don't act like they are fully independent. Changes in a reused template repo seem to get propagated back to our master template. I was willing to deal with repository forks, but these are even worse, acting like clones. I would be quite as happy if all connection to master templates was completely eliminated. I can easily live without forks for an assignment template repo.
